message_ix_models/model/water/build.py

```diff
@@ -153,21 +258,70 @@ def get_spec(context: Context) -> Mapping[str, ScenarioInfo]:
             # Sets  to add
             add.set[set_name].extend(config)
 
-    elif context.nexus_set == "cooling":
-        # Merge technology.yaml with set.yaml
-        context["water set"]["cooling"]["technology"]["add"] = context[
-            "water technology"
-        ]["cooling"]
-        # Update the ScenarioInfo objects with required and new set elements
-        for set_name, config in context["water set"]["cooling"].items():
-            # Required elements
-            require.set[set_name].extend(config.get("require", []))
+    # for both cooling and nexus add share contraints for cooling technologies
+    # cat_tec
 
-            # Elements to remove
-            remove.set[set_name].extend(config.get("remove", []))
+    results = {}
+    cat_tec, nodes_cooling = cat_tec_cooling(context)
+    results["cat_tec"] = cat_tec.values.tolist()
+    n = len(nodes_cooling)
+    # Share commodity for urban water recycling
+    shares_cool = [
+        "share_cooling_ot_fresh",
+        "share_cooling_cl_fresh",
+        "share_cooling_air",
+        "share_cooling_ot_saline",
+    ]
+    commodity_cool = ["ot_fresh", "cl_fresh", "air", "ot_saline"]
+    df_share = make_df(
+        "map_shares_commodity_share",
+        shares=shares_cool,
+        type_tec=[
+            "share_cooling_ot_fresh_share",
+            "share_cooling_cl_fresh_share",
+            "share_cooling_air_share",
+            "share_cooling_ot_saline_share",
+        ],
+        mode="M1",
+        commodity=commodity_cool,
+        level="share",
+    ).pipe(broadcast, node_share=nodes_cooling)
+    df_share["node"] = df_share["node_share"]
+    # re order columns like this ['shares', 'node_share', 'node',
+    # 'type_tec', 'mode', 'commodity', 'level']
```

Comment:
```suggestion
    # re-order columns like this:
```
I'd leave out the order as it is the same as two lines below.